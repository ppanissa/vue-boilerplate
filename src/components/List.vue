<template>
  <ul>
    <li v-for="item in items" :key="item">
      {{ item }}
    </li>
  </ul>
</template>

<script>
export default {
  name: 'List',
  props: {
    items: {
      type: Array,
      default: () => [],
    },
  },
};
</script>
