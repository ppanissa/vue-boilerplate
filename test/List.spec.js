import { shallowMount } from '@vue/test-utils';
import List from '@/components/List.vue';

describe('List.vue', () => {
  it('renders li for each item in props.items', () => {
    const items = [ '1', '2' ];
    const wrapper = shallowMount(List, {
      propsData: {
        items,
      },
    });
    expect(wrapper.findAll('li')).toHaveLength(items.length);
    // Create Snapshot
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('matches snapshot', () => {
    const items = [ 'item 1', 'item 2' ];
    const wrapper = shallowMount(List, {
      propsData: { items },
    });
    // Create Snapshot
    expect(wrapper.html()).toMatchSnapshot();
  });
});
