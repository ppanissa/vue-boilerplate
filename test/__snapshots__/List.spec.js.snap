// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue matches snapshot 1`] = `
<ul>
  <li>
    item 1
  </li>
  <li>
    item 2
  </li>
</ul>
`;

exports[`List.vue renders li for each item in props.items 1`] = `
<ul>
  <li>
    1
  </li>
  <li>
    2
  </li>
</ul>
`;
